### ISSUE TITLE
Add a DeletedUser and assign completed task assignments to it when a user is deleted

### ISSUE THREAD

**Issue by cash** [hltcoe/turkle#108]
Currently, if a user is deleted you not only loose the assignments, but the completed status is not updated on the Batch/Task so annotators won't see it.

The DeletedUser will function like the AnonymousUser.

**Comment by cash** [comment]
Maybe convert each user into a deleted user

**Comment by charman** [comment]
Some existing Turk analysis workflows assume that the "WorkerId" field from the CSV files (which, for Turkle CSV files, is the value of the database field `User.id`) is a stable identifier that can be used to uniquely identify a worker across Batches. Keeping the WorkerId field stable across CSV-files/Batches allows the contributions of a particular Worker to be modeled for QA and calibration purposes.

Because the stable WorkerId assumption is important, I don't think we should assign Task Assignments from deleted users to a single DeletedUser.

I believe we need a mechanism to deidentify or decommission user accounts without deleting the User object (and associated TaskAssignment objects). This could be implemented by adding a new Django Admin action:

https://docs.djangoproject.com/en/2.2/ref/contrib/admin/actions/

that overwrites identifying information associated with the User object.

The ability to delete User objects (and associated TaskAssignments) should be restricted if not completely disabled.

**Comment by cash** [comment]
Agreed. I would prefer to keep the username and first and last names and remove their password hash and email address. The reason for this is being able to trace an annotation back to a user in case you later find out an issue with an annotator.

**Comment by cash** [comment]
Current idea is that deleting doesn't remove the user, but does the following:
 * makes them inactive
 * removes username/email - so that the same person can have a new account later
 * sets some field to mark them as archived so that they don't appear in the list of users
 
Would have to change the UserManager to remove archived users.
